runway/hooks/staticsite/auth_at_edge/templates/check_auth/jwks_rsa/client.py

```diff
@@ -0,0 +1,114 @@
+"""Client for handling retrieval of JWKS signing keys."""
+import json
+import logging
+import urllib
+
+# pylint: disable=relative-beyond-top-level
+from .utils import rsa_public_key_to_pem
+
+
+def is_signing_key(key):
+    """Filter to determine if this is a signing key.
+
+    Keyword Args:
+        key (Dict[str, str]): The key
+    """
+    if key.get('kty', '') != 'RSA':
+        return False
+    if not key.get('kid', None):
+        return False
+    if key.get('use', '') != 'sig':
+        return False
+    return key.get('x5c') or (key.get('n') and key.get('e'))
+
+
+class JwksClient:
+    """Client responsible for retrieval of JWKS signing keys."""
+
+    def __init__(self, options=None):
+        """Initialize.
+
+        Keyword Args:
+            options (Optional[Dict[str, str]]): Options for the client
+        """
+        self.options = options
+        self.logger = logging.getLogger('__file__')
```

Comment:
```suggestion
        self.logger = logging.getLogger(__name__)
```
this can be a bit easier for tracking down where a log message came from as it will give the import path instead of only the file name (without path). it also lets you control the log level from a point higher up in a python module